lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -2013,73 +2022,95 @@ defmodule LightningWeb.WorkflowLive.Edit do
     }
   end
 
-  defp maybe_show_manual_run(socket) do
-    case socket.assigns do
-      %{selected_job: nil} ->
-        socket
-        |> assign(
-          manual_run_form: nil,
-          selectable_dataclips: []
-        )
-
-      %{selected_job: job, selection_mode: "expand"} = assigns
-      when not is_nil(job) ->
-        dataclip =
-          assigns[:follow_run] &&
-            get_selected_dataclip(assigns[:follow_run], job.id)
+  defp maybe_show_manual_run(%{assigns: %{selected_job: nil}} = socket) do
+    assign(socket,
+      manual_run_form: nil,
+      selectable_dataclips: []
+    )
+  end
 
-        body =
-          new_manual_run_form_body(
-            assigns.manual_run_form,
-            job,
-            dataclip
-          )
+  defp maybe_show_manual_run(
+         %{
+           assigns: %{
+             selected_job: selected_job,
+             selection_mode: "expand",
+             current_user: current_user,
+             manual_run_form: manual_run_form,
+             project: project,
+             workflow: workflow,
+             follow_run: follow_run
+           }
+         } = socket
+       ) do
+    dataclip = follow_run && get_selected_dataclip(follow_run, selected_job.id)
+
+    body =
+      new_manual_run_form_body(
+        manual_run_form,
+        selected_job,
+        follow_run,
+        dataclip
+      )
 
-        changeset =
-          WorkOrders.Manual.new(
-            %{dataclip_id: dataclip && dataclip.id, body: body},
-            project: socket.assigns.project,
-            workflow: socket.assigns.workflow,
-            job: socket.assigns.selected_job,
-            user: socket.assigns.current_user
-          )
+    changeset =
+      %{
+        project: project,
+        workflow: workflow,
+        job: selected_job,
+        user: current_user
+      }
+      |> then(fn attrs ->
+        attrs =
+          if body == "{}" do
+            Map.put(attrs, :created_by, current_user)
```

Comment:
Why are we adding the `created_by` only if `body == {}` ? 
I can see in the above block we are already passing in a `user: current_user`, is it any different from `created_by`?

Reply:
When the user accepts the suggestion of the empty body, the user is the creator of the dataclip. This is not needed when the user types the input but without typing it is.
I would refactor the existing validation that checks about this `created_by` but once this is a huge module that handle many cases I am leaving it do it after some prioritization.